src/sensai/vector_model.py

```diff
@@ -247,24 +259,44 @@ def predict(self, x: pd.DataFrame) -> pd.DataFrame:
     def _predict(self, x: pd.DataFrame) -> pd.DataFrame:
         pass
 
-    def fit(self, X: pd.DataFrame, Y: pd.DataFrame):
+    def fit(self, X: pd.DataFrame, Y: pd.DataFrame, fitPreprocessors=True, fitTargetTransformer=True):
         """
         Fits the model using the given data
 
         :param X: a data frame containing input data
         :param Y: a data frame containing output data
+        :param fitPreprocessors: if False, the model's feature generator and input transformer will not be fitted.
+            If a preprocessor requires fitting, was not separately fit before and this option is set to False,
+            an exception will be raised.
+        :param fitTargetTransformer: if False, the model's target transformer will not be fitted.
+            If it requires fitting, was not separately fit before and this option is set to False,
+            an exception will be raised.
         """
         log.info(f"Training {self.__class__.__name__}")
         self._predictedVariableNames = list(Y.columns)
-        X = self._computeInputs(X, Y=Y, fit=True)
+        X = self._computeInputs(X, Y=Y, fit=fitPreprocessors)
         if self._targetTransformer is not None:
-            self._targetTransformer.fit(Y)
-            Y = self._targetTransformer.apply(Y)
+            if fitTargetTransformer:
+                Y = self._targetTransformer.fitApply(Y)
+            else:
+                Y = self._targetTransformer.apply(Y)
         self._modelInputVariableNames = list(X.columns)
         self._modelOutputVariableNames = list(Y.columns)
         log.info(f"Training with outputs[{len(self._modelOutputVariableNames)}]={self._modelOutputVariableNames}, inputs[{len(self._modelInputVariableNames)}]=[{', '.join([n + '/' + X[n].dtype.name for n in self._modelInputVariableNames])}]")
         self._fit(X, Y)
-        self._isFitted = True
+        self.__modelIsFitted = True
+
+    # TODO: just showing how I roughly imagine the interface in the future.
+    #  Pre- and postprocessors can typically not be fit on batches which is one of the reasons I added the flag above.
+    def fitDataSet(self, dataSet):
+        """
+        Fits the models on a DataSet which does not need to be loaded to ram. Any pre- and postprocessors forming
+        part of the model are expected to be fit prior to calling this method or to not require fitting.
+        :param dataSet:
+        :return:
+        """
+        for batch in dataSet:
+            self.fit(batch.X, batch.Y, fitPreprocessors=False, fitTargetTransformer=False)
```

Comment:
This won't work. We can't just call fit several times. The learning algorithm will usually want to do the iteration over batches.

Reply:
You are right, I didn't think this through. I will just remove it for now, the work will happen in another PR. Do you have an idea for how to realize it without changing the model implementations? I don't see a straightforward way now, but it might be because its late and I am tired. The only thing coming to mind is to do some kind of aggregation of batches before we pass it to the model but it partially defeats the purpose of having a generator...

Reply:
Feel free to write me on wire, the discussion is out of scope for this PR